### DISCUSSION TITLE
新的Task Manager思路

### DISCUSSION THREAD

**Discussion by jc3213** [jc3213/download_with_aria2#47] [Ideas]
测试链接：https://jc3213.github.io/jslib/manager/index.html

不知道是否应该把现有的任务详细页面改成这种形式？

@ivysrono @wordpure @daiaji

**Comment by daiaji** [comment]
实在没有美术，就完全无所谓了。
老实说没感觉有区别。
然后firefox新下载按钮点了没用。
bug回归？

**Reply by jc3213** [reply]
窗口化的话需要拉到至少1400的宽度，800的高度才行，这个是跟窗口生成有关。我记得4.0.0开始我没改过创建window的这一块。

Chromium系列的话这个限制要小些，估计还是跟`chrome.windows.getCurrent()`有关。但是我从console看，firefox算出来的数据是没错的，但是莫名其妙的新建窗口在屏幕正中。因为宽度是680x780，如果这个范围不在Firefox窗口内，就不会正常显示。当然，范围超出浏览器后Chromium也不会正常显示。

**Reply by jc3213** [reply]
发现了，貌似Firefox在`chrome.windows.create`的时候，所有参数必须是整数，而不能是浮点，否则会报错。这里我想办法修复一下。

**Reply by jc3213** [reply]
对比了下4.0.0的changes，发现貌似中途我复制粘贴的时候把`lib/tools.js`给弄错了……导致后续的更新有这个bug。不过已经修复了。OTL

**Reply by daiaji** [reply]
修好了？

**Reply by jc3213** [reply]
是的，理论上4.0.1.1504就已经修好了

**Reply by daiaji** [reply]
Ok，我数据又寄了，回去试试看。

**Comment by ivysrono** [comment]
对界面没有特别想法。
之前那个仓库删掉了？里面的issue找不到了？似乎一些老问题又冒出来了。
但似乎结果又与原来不一样，比方说网易邮箱附件成了`readdata.jsp`

**Reply by jc3213** [reply]
大概是哪个版本出现的？可以的话说下哪个API，然后尽量提供一下`console.log`，毕竟我没有网易邮箱。

理论上只有因为referer这里因为要做额外选项改过一次，本质的行为逻辑跟老的一模一样才对啊？

相关修改 https://github.com/jc3213/download_with_aria2/compare/3.21.7...4.0.0.1480#diff-f8327d456eb627712c0d419126894ea390b29d00c00b7727ad53e6fffbc17aa0L89-R109

**Reply by jc3213** [reply]
我用`webRequest`，然后测试QQ邮箱附件，提示还算正常。希望能提供例子做参考。我也添加了`console.log`，试试最新的代码吧。另外，如果是`downloads`出错的话，我估计是没法修复的。

**Reply by ivysrono** [reply]
log:
`attachment; filename="=?UTF8?B?572R5piT6YKu566x5paw5bu64oCc5paH?= =?UTF8?B?5pys4oCd5paH5qGj55qE5paH5Lu25ZCN54m5?= =?UTF8?B?5Yir6ZW/5Y+v6IO95Lya5Y+q5oiq5Y+W5YW25Lit5LiA6YOo5YiGLnR4dA==?=" 网易邮箱新建“文本”文档的文件名特别长可能会只截取其中一部分.txt`
webRequest 和 downloads API 都是这个提示，下载下来都是jsp文件。

**Reply by jc3213** [reply]
不应该啊，文件名正确的解码了。而且传递的是out参数啊？应该是后续部分出问题了。我研究下

**Reply by jc3213** [reply]
找到错误了，`downloads`的处理有问题，修改了`function`的需要的变量，但是没修改传递的变量。但是`webRequest`没发现问题
![1](https://user-images.githubusercontent.com/8744936/204066029-de79d83b-f0c1-4a8d-8a57-cfc03c9eef04.png)
![2](https://user-images.githubusercontent.com/8744936/204066031-6824391c-b4a2-4e4a-bb9d-777e4a6d0e8b.png)

**Reply by jc3213** [reply]
理论上`downloads` API的修复了，`webRequest`我添加了一个对options的log，你看看效果？可以的话下次反馈开个新回复

**Reply by ivysrono** [reply]
两种API在1504均已修复网易邮箱。
